.github/workflows/test_editable_pip_install.yml

```diff
@@ -0,0 +1,63 @@
+name: Test Editable Pip Install
+
+on:
+  push:
+    branches:
+      - 'main'
+  pull_request:
+    types: [opened, synchronize]
+
+jobs:
+  editable_install:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-20.04, macos-latest]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      # This shouldn't be necessary, but for some reason it is on macos
+      - name: install dev_requirements.txt
+        if: ${{ matrix.os == 'macos-latest' }}
+        run: python3 -m pip install -r dev_requirements.txt
```

Comment:
We do want to do this in CI even if it's not required for a user install, since we need things to be pinned - I think it's also worth figuring out which package exactly macos requires from here, but that can wait.  But at least we should update the comment to reflect that we don't want to remove this entirely once we figure that out

Reply:
I see. I understand the purpose of `dev_requirements.txt` a lot better now (for some reason I thought it was automatically read and installed by pip during the `pip install .` process).

And macos requires a bunch of things, from cmake to argh and the like.